clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackend.java

```diff
@@ -289,11 +289,17 @@ void executeCommandWithRetry(
         return;
       }
       LOGGER.info("Sending command {} of topic: {}, partition: {} to fork process.", command, topicName, partition);
+      if (command.equals(START_CONSUMPTION)) {
+        /**
+         * StartConsumption operation may take long time to wait for non-existence store/version until it times out.
+         * Add version check here so that the long waiting period won't impact forked ingestion process Netty server
+         * performance.
+         */
+        Utils.waitStoreVersionOrThrow(topicName, getStoreIngestionService().getMetadataRepo());
```

Comment:
What if this method takes up to 30 sec, how does it effect the Helix threads? Would it also impact other Helix threads and even block some state transitions?

Reply:
The Helix implementation maintains a large thread pool (configurable, right now 450 in EI), so it won't block, and that's how non-II code handles it today.